### PR TITLE
FOR-18 pose meter to pixel

### PR DESCRIPTION
**Pull request by CasLeeflang**

### CODE REVIEW

File: formicarium/formicarium/DiffRobot.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from geometry_msgs.msg import Twist
 from sensor_msgs.msg import LaserScan
 from rclpy.node import Node
+import pygame
```
Comment (CrazyPandabier) [inline formicarium/formicarium/DiffRobot.py]:
```suggestion
from pygame import Rect
```
dont need to import the complete library only Rect is enough

File: formicarium/formicarium/DiffRobot.py [hunk 2]
```diff
@@ -43,6 +44,8 @@ def __init__(self,name:str, wheel_radius: float, wheel_base: float, start_x: flo
         self.wheel_base = wheel_base
         self.wheel_radius = wheel_radius
         self.rect = self.image.get_rect(center=(self.x, self.y))
+        self.hitbox = pygame.Rect(0, 0, 69.5, 69.5)
```
Comment (CrazyPandabier) [inline formicarium/formicarium/DiffRobot.py]:
```suggestion
        self.hitbox = Rect(0, 0, 69.5, 69.5)
```
here you can then just use Rect

File: formicarium/formicarium/Environment.py [hunk 3]
```diff
@@ -31,14 +31,14 @@ def add_robot(self, robot: IRobot) -> None:
         self.robot_names.append(robot.name)
         self.robot_group.add(robot)
     
-    def add_obstacle(self, dim_pos) -> None:
-        _obstacle = pygame.Rect(dim_pos)
+    def add_obstacle(self, pos_dim) -> None:
+        _obstacle = pygame.Rect(pos_dim)
```
Comment (CrazyPandabier) [inline formicarium/formicarium/Environment.py:33]:
```suggestion
        _obstacle = Rect(pos_dim)
```

File: formicarium/formicarium/Environment.py [hunk 4]
```diff
@@ -49,12 +49,14 @@ def check_collision_lidar(self, x: float, y: float) -> bool:
     def check_collision_robot(self, robot:sprite.Sprite) -> bool:
         _robot_group = self.robot_group.copy()
         _robot_group.remove(robot)
-        if pygame.sprite.spritecollideany(robot, _robot_group or []):
-            return True
+        for _robot in _robot_group:
+            if pygame.Rect.colliderect(robot.hitbox, _robot.hitbox):
```
Comment (CrazyPandabier) [inline formicarium/formicarium/Environment.py:51]:
```suggestion
            if Rect.colliderect(robot.hitbox, _robot.hitbox):
```

File: formicarium/formicarium/Environment.py [hunk 5]
```diff
@@ -49,12 +49,14 @@ def check_collision_lidar(self, x: float, y: float) -> bool:
...
+                return True
+            
+        for _obstacle in self.obstacle_group:
+            if pygame.Rect.colliderect(robot.hitbox, _obstacle):
```
Comment (CrazyPandabier) [inline formicarium/formicarium/Environment.py:55]:
```suggestion
            if Rect.colliderect(robot.hitbox, _obstacle):
```

**Review by CrazyPandabier — COMMENTED** [review]
I made some suggestions for the pygame.Rect.

File: formicarium/formicarium/DiffRobot.py (hunk 1 above)
Comment (CasLeeflang) [reply formicarium/formicarium/DiffRobot.py]: That is correct, it is changed and Rect() is directly called instead of pygame.Rect for the hitbox property

**Review by CrazyPandabier — APPROVED** [review]

**Pull request merged**